src/Renderer/ThreeExtended/FeaturesUtils.js

```diff
@@ -0,0 +1,146 @@
+
+        if (scalar >= -epsilon && scalar <= norm + epsilon) {
+            const distance = Math.abs(y21 * x0 - x21 * y0 + x2 * y1 - y2 * x1) / norm;
+            if (distance <= epsilon) {
+                return true;
+            }
+        }
+    }
+
+    return false;
+}
+
+function getClosestPointsIndice(point, points, epsilon) {
+    const x0 = point._values[0];
+    const y0 = point._values[1];
+    let squaredEpsilon = epsilon * epsilon;
+    let closestPoint;
+    for (var i = 0; i < points.length; ++i) {
+        const x1 = points[i]._values[0];
+        const y1 = points[i]._values[1];
+        const xP = x0 - x1;
+        const yP = y0 - y1;
+        const n = xP * xP + yP * yP;
+        if (n < squaredEpsilon) {
+            closestPoint = points[i];
+            squaredEpsilon = n;
+        }
+    }
+    return closestPoint;
+}
+
+function pointIsInsidePolygon(point, polygonPoints) {
+    // ray-casting algorithm based on
+    // http://www.ecse.rpi.edu/Homepages/wrf/Research/Short_Notes/pnpoly.html
+
+    const x = point._values[0];
+    const y = point._values[1];
+
+    let inside = false;
+    // in first j is last point of polygon
+    // for each segment of the polygon (j is i -1)
+    for (let i = 0, j = polygonPoints.length - 1; i < polygonPoints.length; j = i++) {
+        const xi = polygonPoints[i]._values[0];
+        const yi = polygonPoints[i]._values[1];
+        const xj = polygonPoints[j]._values[0];
+        const yj = polygonPoints[j]._values[1];
+
+        // isIntersect semi-infinite ray horizontally with polygon's edge
+        const isIntersect = ((yi > y) != (yj > y))
+            && (x < (xj - xi) * (y - yi) / (yj - yi) + xi);
+        if (isIntersect) {
+            inside = !inside;
+        }
+    }
+
+    return inside;
+}
+
+
+function getFeatureAtCoordinate(coordinate, type, coordinates, epsilon, properties, result) {
+    if (type == 'linestring' && pointIsOverLine(coordinate, coordinates, epsilon)) {
+        result.lines.push({ properties, coordinates });
+    } else if (type == 'polygon' && pointIsInsidePolygon(coordinate, coordinates)) {
+        result.polygons.push({ properties, coordinates });
+    } else if (type == 'point') {
+        const closestPoint = getClosestPointsIndice(coordinate, coordinates, epsilon);
+        if (closestPoint) {
+            result.points.push({ properties, coordinates: closestPoint });
+        }
+    }
+}
+export default {
+    /**
+     * filters the features that are under the coordinate
+     *
+     * @param      {Coordinates}  coordinate  the coordinate for the filter condition
+     * @param      {Features}  features  features to filter
+     * @param      {number}  epsilon  tolerance around the coordinate (in coordinate's unit)
+     * @return     {array}  array of filters features
+     */
+    getFeaturesAtCoordinate(coordinate, features, epsilon = 0.1) {
+        const result = { points: [], lines: [], polygons: [] };
```

Comment:
I don't see the point of returning the features grouped by geometry types?
Why don't you simply return a list of matching features?

(and please do not force push in the middle of a review, it makes it hard to have a discussion because previous comments are hidden)

Reply:
And `getXXX` name scheme is reserved for getter imho. `extract`? `filter`?

Reply:
Hmmm... You're not looping over `features` so I'm really wondering: what is the intent of this fonction? 
Can a user pass an array of features?
Or are only you extracting geometry/geometries that match a given coordinate from a **single** feature?

Reply:
> I don't see the point of returning the features grouped by geometry types?

Ok, for a list of typed features

> You're not looping over features

I loop over all features through, features.geometries have coordinates and features.features have properties [see here](https://github.com/iTowns/itowns/pull/470/files/9d7cb65a3791c5a3d7889563079740772cfddb06#diff-489ba48df1c76dca6a092ba77ffc0b64R133)

> And getXXX name scheme is reserved for getter imho. extract? filter?

ok for filter